scenes/lessons/7th_class/miandam/miandam.py

```diff
@@ -0,0 +1,578 @@
+        is_not_miandam = Tex('Միանդամներ չեն', tex_template=ARMTEX)
+        is_not_miandam.align_to(is_miandam, UP).shift(RIGHT * 3.5)
+
+        not_miandam_1 = MathTex('a', ' + ', 'b')
+        not_miandam_1[1].set_color(RED)
+        not_miandam_2 = MathTex(r'\frac{5}{a}\cdot x')
+
+        not_miandams = VGroup(not_miandam_1, not_miandam_2)
+        not_miandams.arrange(DOWN, buff=0.5).next_to(is_not_miandam, DOWN, buff=0.5)
+
+        surr_rects_miandam_4 = VGroup(
+            SurroundingRectangle(miandams_4[0]),
+            SurroundingRectangle(miandams_4[2])
+        )
+
+        surr_rect_not_miandam_2 = Rectangle(color=YELLOW)
+        surr_rect_not_miandam_2.move_to(not_miandam_2).shift([-0.35, -0.4, 0]).scale(0.11).stretch(1.75, 1)
+        not_miandam_haytarar = MathTex(r'a', color=RED).move_to(not_miandam_2).shift([-0.35, -0.4, 0])
+
+
+# ANIMATIONS
+
+        self.play(Write(sahmanum[0], run_time=2.5))
+        self.play(Write(sahmanum[1], run_time=1.5))
+        self.play(Create(surr_rect_sahmanum))
+        self.wait(0.25)
+
+        self.play(Write(is_miandam))
+        self.wait(0.25)
+
+        self.play(Write(miandams_1[0]))
+        self.wait(0.25)
+        self.play(Write(miandams_1[1], run_time=0.5))
+        self.play(Write(miandams_1[2]))
+        self.wait(0.75)
+
+        self.play(
+            AnimationGroup(
+               Write(miandams_2[0]),
+               Write(miandams_2[1]),
+               lag_ratio=0.75, run_time=0.75
+            )
+        )
+        self.wait(0.1)
+        self.play(
+            AnimationGroup(
+               Write(miandams_2[2]),
+               Write(miandams_2[3]),
+               lag_ratio=0.75, run_time=0.75
+            )
+        )
+        self.wait(0.1)
+        self.play(Write(miandams_2[4], run_time=2, rate_func=linear))
+        self.wait()
+
+        self.play(Write(miandams_3[0], run_time=1.75, rate_func=linear))
+        self.wait()
+        self.play(Write(miandams_3[1]))
+        self.play(Write(miandams_3[2]))
+        self.wait(0.25)
+
+        self.play(Write(is_not_miandam))
+        self.wait(0.25)
+
+        self.play(Write(not_miandam_1))
+        self.wait()
+
+        self.play(Write(miandams_4))
+        self.wait()
+
+        self.play(Write(not_miandam_2))
+        self.wait()
+
+        self.play(Create(surr_rects_miandam_4))
+        self.wait(0.5)
+        self.play(FadeOut(surr_rects_miandam_4))
+        self.wait(0.5)
+
+        self.play(
+            Create(surr_rect_not_miandam_2),
+            FadeIn(not_miandam_haytarar)
+        )
+        self.wait(0.5)
+        self.play(FadeOut(surr_rect_not_miandam_2))
+        self.wait()
+
+        self.remove(*self.mobjects)
+
+
+class Miandam(Scene):
+    def construct(self):
+        RoadScene.construct(self)
+        FiveXPlusSixY.construct(self)
+        FiveXPlusSixYPlusFourZ.construct(self)
+        SahmanumOrinakner.construct(self)
```

Comment:
I think having 4 functions, that take `Scene` as an argument would be much more explicit, than having 4 Scene classes, and calling their method with different class's `self`. What I recommend essentially is replacing
```python
class A(Scene):
    def construct(self):
        pass
```
with
```python
def construct_a(scene):
    pass
```

Also, I would recommend to clear the scene (if necessary) after each sub-scene just in case.

Reply:
ամեն սինի վերջում ամեն ինչ ֆեյդաութ եմ անում, բայց իրականում, մեծ հաշվով, էս վերջին սինը պետք էլ չի, մեկ ա մոնտաժի ժամանակ կտրտում կպցնում եմ ոնց պետք ա լինելու։ ինձ ավելի հարամր ա 4 հատ սին ունենամ, տենց հատ-հատ անելով գնամ։ էս առաջարկդ չհասկացա ոնց ա աշխատում, construct-ը մանիմական բան ա, ռենդռեռը հենց Scene կլասսի construct անունով ֆունկցիան ա նայում ու սկսում ռենդել

Reply:
հա, գիտեմ, կմտածեի վերջինը մենակ ռան կէնես, էդ պատճառով կսեի վերևինները կռնային class չէղնեին։ Եթե վերևիններն էլ ես ռան էրել, ճիշտ կէնես։